datarepo-clienttests/src/main/java/runner/TestRunner.java

```diff
@@ -0,0 +1,326 @@
+package runner;
+
+import bio.terra.datarepo.client.ApiClient;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import io.kubernetes.client.openapi.models.V1Deployment;
+import io.kubernetes.client.openapi.models.V1Pod;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import runner.config.TestConfiguration;
+import runner.config.TestScriptSpecification;
+import runner.config.TestUserSpecification;
+import utils.AuthenticationUtils;
+import utils.FileUtils;
+import utils.KubernetesClientUtils;
+
+class TestRunner {
+
+  TestConfiguration config;
+  List<TestScript> scripts;
+  List<ThreadPoolExecutor> threadPools;
+  List<List<Future<UserJourneyResult>>> userJourneyFutureLists;
+  Map<String, ApiClient> apiClientsForUsers; // testUser -> apiClient
+  // TODO: have ApiClients share an HTTP client, or one per each is ok?
+
+  static long secondsToWaitForPoolShutdown = 60;
```

Comment:
do we have a const across files for this? Is that overkill?

Reply:
There's no cross-file constant for this. I put it here because this is the only place it's used, and I don't expect it to be changed frequently or per-test. There are similar constants in some of the utils classes (see DataRepoUtils and KubernetesClientUtils). If it gets used in other places, then I agree that pulling it out into a separate enum or something would be better.